unime/src/routes/(app)/scan/+page@.svelte

```diff
@@ -20,73 +21,60 @@
   import BottomNavBar from '$src/lib/components/molecules/navigation/BottomNavBar.svelte';
   import { state } from '$src/stores';
 
+  import CameraSlash from '~icons/ph/camera-slash';
+
   let scanning = false;
-  let permissions: 'granted' | 'denied' | 'prompt' | undefined = undefined;
+  // We temporarily introduce this type that extends `PermissionState` to handle a possible error when checking for permissions.
+  let permissions_nullable: PermissionState | null;
```

Comment:
maybe we can call this optionalPermissions, so we keep the terminology of rusts `Option` and typescript a bit similar